Applications/Spire/Source/SpireTester/ArrayListModelTester.cpp

```diff
@@ -0,0 +1,273 @@
+#include <doctest/doctest.h>
+#include "Spire/Ui/ArrayListModel.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Spire;
+
+TEST_SUITE("ArrayListModel") {
+  TEST_CASE("push") {
+    auto model = ArrayListModel();
+    REQUIRE(model.get_size() == 0);
+    auto signal_count = 0;
+    auto connection = scoped_connection(model.connect_operation_signal(
```

Comment:
Unit test by using a ```std::deque<Operation>```, push operations onto the queue and then assert the contents of the queue:

      auto operations = std::deque<TableModel::Operation>();
      translation.connect_operation_signal([&] (const auto& operation) {
        operations.push_back(operation);
      });
      ...
      REQUIRE(operations.size() == 1);
      auto operation = operations.front();
      operations.pop_front();
      REQUIRE(operation == ...);